libyul/optimiser/Suite.cpp

```diff
@@ -366,8 +364,14 @@ map<string, unique_ptr<OptimiserStep>> const& OptimiserSuite::allSteps()
 			SSATransform,
 			StructuralSimplifier,
 			UnusedPruner,
-			VarDeclInitializer,
-			VarNameCleaner
+			VarDeclInitializer
+
+			// NOTE: VarNameCleaner implements the same interface but it's not a real optimisation
```

Comment:
I think we should not keep this comment here. Maybe better to have a very short list of the steps that are not part of this list and then explain in the class of that step why it is not a "regular" step.

Reply:
I put it here to prevent someone from adding the step back thinking it's just an accidental omission (insipired by me wondering the the same thing about `UnconditionalSimplifier` missing from `yulopti :)).

I see you did leave a short note there after all. I think it's a good practice to leave hints like that when possible. In moderation of course because they do go out of date/out of sync easily.